langlib/lang.table/src/main/ballerina/table.bal

```diff
@@ -123,6 +189,20 @@ public isolated function 'map(table<MapType> t, @isolatedParam function(MapType
 #
 # The parameter `func` is applied to each member of parameter `t`.
 #
+# ```ballerina
+# table<record {|string name; int? salary;|}> employees = table [
+#     {name: "Jo", salary: 1200},
+#     {name: "Emma", salary: ()}
+# ];
+# 
+# employees.forEach(isolated function(record {|string name; int? salary;|} employee) {
+#     if employee.salary == () {
+#         employee.salary = 1000;
+#     }
+# });
+# employees ⇒ [{"name":"Jo","salary":1200},{"name":"Emma","salary":1000}]
```

Comment:
Since we are simplifying examples, we can compute the total salary or something here, similar to the array langlib. The anonymous function will no longer be isolated but that should be OK.